### PR TITLE
added grant_type support to client resource

### PR DESCRIPTION
**Pull request by domg123**

### CODE REVIEW

**Comment by alexkappa** [comment]
Hi @domg123, thanks for submitting this! Looking at the errors, it might be that `d.Set("grant_types", c.GrantTypes)` didn't work as expected.

[This error message](https://app.wercker.com/Yieldr/terraform-provider-auth0/runs/build/5b3d63988359820001241991?step=5b3d63ad37b4370001ff514b) suggests that the `grant_types` are being set to an empty array after the value is being read back from Auth0:

```
grant_types.#: "4" => "0"
```

Sometimes the Auth0 API doesn't return all the values from a resource, unless you explicitly request them. We might need to update the Auth0 SDK for this to work.

**Comment by alexkappa** [comment]
Ok. So I've pulled your branch to verify my hypothesis, but it turns out I was wrong :)

The solution was much easier, it was related to `grant_types` being defined as optional, but not as computed. In this case, since the `grant_types` will be generated if none were set (by Auth0).

So terraform was confused as our expected was no `grant_types` while we actually had an array of 4 items.

TLDR; we just need a [`Comuted: true`](https://github.com/yieldr/terraform-provider-auth0/blob/domg123-feature/client-enabled-grants/auth0/resource_auth0_client.go#L69) to the schema and it should be good to go!

**Comment by domg123** [comment]
Thanks for taking a look!  I've added Computed.

**Comment by alexkappa** [comment]
Thank you for contributing!
